### PR TITLE
[ci] Fix apache/pulsar-test-infra/paths-filter action permission in CodeQL workflow

### PR DESCRIPTION
**Pull request by yaalsn**
### Motivation

The codeql workflow needs read pull-requests permission when use action `apache/pulsar-test-infra/paths-filter@master`
image

### CODE REVIEW

**Review by shoothzj — COMMENTED** [review]
Could you please share the fail link, in my memory, CodeQL workflow always run well.

**Comment by hangc0276** [comment]
> Could you please share the fail link, in my memory, CodeQL workflow always run well.

@shoothzj We run the CI in our own repo and found the CodeQL doesn't have permission

**Review by hangc0276 — APPROVED** [review]

**Review by shoothzj — CHANGES_REQUESTED** [review]
IMO, it's a little hard to maintaince this change, the community doesn't face this problem, at least we should have some **CodeQL** documentaion.
I see the original permissions is copied from https://github.com/github/codeql-action/blob/59c62518002b2015b741cd192d3b36bc7787efc7/README.md.

Or we can test it in a fork repo? I can help with this :)

**Comment by eolivelli** [comment]
Is this change exposing secrets to github users?

@yaalsn if you want to maintain a provate fork  I think that you have already other changes that are no here in the public repo.
If this change add security risky for the community I suggest to push this change only to your repo

**Comment by shoothzj** [comment]
@yaalsn Thanks for your explanation, I change my mind to +0, I won't block this PR. But it seems werid you still needs Approval to triggered from CI. You already have pr merged into repo.

**Review by nicoloboschi — APPROVED** [review]

**Review by shoothzj — APPROVED** [review]

**Pull request merged**
